git/deploy_key.go

```diff
@@ -0,0 +1,101 @@
+package git
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+func newDeployKey(g *Github, o owner, r repo, key string) (*deployKey, error) {
+	k := deployKey{
+		Github: g,
+		owner:  o,
+		repo:   r,
+		key:    key,
+	}
+	// Set the key values from the remote
+	err := k.Get()
+	if err != nil {
+		return &k, err
```

Comment:
Yeah, Get isn't quite the right name. It has no return values. Will change name.